src/ethereum/fees/ethMiningFees.ts

```diff
@@ -253,23 +256,26 @@ export const calcOptimismRollupFees = (
     throw new Error('Invalid rawTx string')
   }
 
-  let rawTxCost = 0
+  let txCompressedSize = 0
   for (let i = 0; i < unsignedRawTxBytesArray.length; i++) {
     if (unsignedRawTxBytesArray[i] === '00') {
-      rawTxCost += 4 // cost for zero byte
+      txCompressedSize += 4 // cost for zero byte
     } else {
-      rawTxCost += 16 // cost for non-zero byte
+      txCompressedSize += 16 // cost for non-zero byte
     }
   }
 
-  const gasUsed = add(
-    add(rawTxCost.toString(), fixedOverhead),
-    MAX_SIGNATURE_COST
-  )
+  txCompressedSize = (txCompressedSize + MAX_SIGNATURE_COST) / 16
```

Comment:
So the spec postpones the division to the very end, but I see that we are using biggystring's floating-point support to shuffle this earlier, while still getting the same result. I guess that works.

Reply:
I got it mixed up because I used two sources which had the math different as you described. I'll make it match the commented source.